lib/todo_web/live/item_live/item_component.html.heex

```diff
@@ -1,7 +1,18 @@
 <div id={"item-#{@item.id}"} class='flex flex-col w-1/2 md:w-1/3 lg:w-1/4 m-2 border-solid border-2 border-black rounded-md'>
   <div class='m-2'>
     <div><%= @item.description %></div>
-    <div class='mt-1'><strong>Due</strong> <%= Todo.Date.pretty_date(@item.due_date) %></div>
+    <div class='mt-1'>
+      <strong>Due</strong>
+      <%= if ItemComponent.compare_date(@item.due_date) === 0  do %>
+        <span class="due-current"> <%= Todo.Date.pretty_date(@item.due_date) %></span>
+      <% end %>
+      <%= if ItemComponent.compare_date(@item.due_date) === -1 do %>
+        <span class="due-past"> <%= Todo.Date.pretty_date(@item.due_date) %></span>
+      <% end %>
+      <%= if ItemComponent.compare_date(@item.due_date) === 1 do %>
+        <span> <%= Todo.Date.pretty_date(@item.due_date) %></span>
+      <% end %> 
```

Comment:
You've got the right idea here, but this code will need to call the `ItemComponent.compare_date/1` function three times on each render. We should only need to compare the dates once. My suggestion would be to return the class names directly from the comparison function itself. Then you could have something like this:

```heex
<span class={ItemComponent.due_date_styles(@item.due_date)}>
  <%= Todo.Date.pretty_date(@item.due_date) %>
</span>
```

This way, you can compare the dates just like you are currently doing in the `ItemComponent.compare_date/1` function but just return the styles as a result of the comparison. Of course, when the due date is sometime in the future, you could just return an empty string so that no extra styles are added.